CMakeLists.txt

```diff
@@ -1,36 +1,23 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.12.0)
```

Comment:
It's been a while since I used cmake.  What changes below are prompting the version bump?

(It's beneficial to have the version as low as possible for the widest array of compatibility)

Reply:
The advantages are that target_link_libraries becomes more intelligent (does proper including and linking of other libraries to the target without the need for adding those library paths to target_include_directories)

Really anything above CMake version 3.0 can do that. But 2.8.3 cannot.